src/components/ComputeModal.js

```diff
@@ -572,16 +599,18 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
         const isDataproc = (sparkMode, runtimeConfig) => !sparkMode && !runtimeConfig?.diskSize
         const runtimeConfig = currentRuntimeDetails?.runtimeConfig
         const gpuConfig = runtimeConfig?.gpuConfig
-
-        setSparkMode(runtimeConfig?.cloudService === cloudServices.DATAPROC ?
+        const newSparkMode = runtimeConfig?.cloudService === cloudServices.DATAPROC ?
```

Comment:
I really dislike nested ternaries. Can we make this into a `Utils.cond` so the logic is easier to parse?